### ISSUE TITLE
test(model/v41): independent numeric token-path oracle for V4.1

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12900]
# test(model/v41): independent numeric token-path oracle for V4.1

```routing
lane: model
paths: ["internal/model/v41_oracle_test.go", "internal/model/testdata/v41_oracle_vectors.json"]
expected_steps: 6
repo: fak
public_issue: anthony-chaudhary/fak#12641
```

## Parent context
Parent: #12641 (independent numeric oracle for the V4.1 token path).

## Current state
Implementing this leaf's through-line is not yet started; the parent track is open.

## Problem frame
- Centrality: Core ? independent oracle gates every forward claim.
- P1: advanced - independent oracle gates every forward claim.
- P2: advanced - prevents self-referential test success.
- P3: preserved - V4-Flash oracle superseded separately.
- P4: advanced - oracle does not reuse production helpers.

## Why now
This leaf is on the critical path for native V4.1 execution and its parent track is open.

## Core through-line
Build an independent CPU oracle for the pinned V4.1 equations (revision
`dba1be0a40aa45a94ad051997016db3960a90277`), replacing the V4-Flash-0731 scope of `fak#12641`.
Cover mHC four-stream mixing, CED/CSA2 shared-KV attention, Engram row retrieval, the 384/top-6
router with shared expert, FP8 32x32-block and MXFP4 decoding, and final norm/head logits.

## Working spine
Intake -> bounded change -> focused witness -> landed receipt.

## Gold-plating boundary
The test must not reuse production scalar helpers. No production flag or hidden marker that
bypasses checkpoint admission. A reduced fixture proves component execution, not full-checkpoint
token parity.

## Witness
```bash
go -C ..\fak test ./internal/model -run 'TestV41Oracle' -count=1
```

### File:Line Seam
`internal/model/v41_oracle_test.go` (new; independently derived expected values)

### Blast Radius & Affected Lanes
- **Primary Lane:** `model`
- **Unaffected Lanes:** production code untouched by this leaf

### Quarantined Fallback Mechanism
Full published checkpoints continue to fail closed at the first unimplemented stage rather than
falling through to generic attention.

## Scoped Acceptance Criteria
- [ ] [SW-VERIFIED] Checked-in expected values are derived independently from the pinned formulas and cite the revision.
- [ ] [SW-VERIFIED] The Go test does not reuse production scalar helpers.
- [ ] [SW-VERIFIED] Reduced synthetic model executes through the native Prefill/Step seam and matches logits.
- [ ] [SW-VERIFIED] Published full checkpoints still fail closed.

## Done condition
The scoped acceptance criteria above pass with the cited witness; no sibling lane regresses.

## Acceptance gate
Focused witness passes, package tests are green, vet and 5-gate boundary checks are clean.

## Work estimate
Estimate: 2 points (bounded leaf).

## Overall completion contribution
Contribution: 2/40 points (provisional baseline).

## Completion standard
development

## Closure binding
The resolving commit cites this issue with Fixes/Ref and carries the repository-qualified lane trailer.

## Dependency marker (set 2026-09-12)

Start blocked by: #12901 (native V4.1 forward seam) and #12895-#12899 (mHC, CED/CSA2, Engram, router, FP8/MXFP4 leaves). This leaf cannot run its native Prefill/Step witness before those land.

Blocked-by record: #12913.

**Comment by anthony-chaudhary** [comment]
﻿Closing as satisfied by already-landed trunk commit `47fe0a6a8883253eea44e65348f9c1a4ee019dac` ("test(model): independent numeric oracle for DeepSeek V4.1 Flash components"), which is an ancestor of `origin/main`.

**Landed artifacts (on `origin/main`):**
- `internal/model/v41_oracle_test.go` (781 lines, independent scalar transcription)
- `internal/model/testdata/deepseek_v41_oracle_expected.json` (3364 lines, expected-values fixture)

**Acceptance witness (run on the landed tree):**
```text
go test ./internal/model -run 'TestDeepSeekV41Oracle' -count=1 -v   => PASS
  TestDeepSeekV41OracleFixtureValues
  TestDeepSeekV41OracleRouteNonVacuous
  TestDeepSeekV41OracleEngramStructure
  TestDeepSeekV41OracleSinkhornIterationCount
  TestDeepSeekV41OracleMHCCombOrientation
  TestDeepSeekV41OracleCompressSchedule
go test ./internal/model -run 'TestV41|TestDeepSeekV41|TestV4' -count=1 => ok 2.415s
go vet ./internal/model                                            => exit 0
go run ./cmd/fak-boundary check --all (fak-private)                => PASS 0 violations, all 5 gates
```

All scoped acceptance criteria are `[SW-VERIFIED]` on fixtures; no physical or full-model generation claim is made. Full native V4.1 execution and physical qualification remain open in #12640.
